### PR TITLE
nixos/borgbackup: Add a persistent option

### PR DESCRIPTION
**Pull request by blargg**
Persistent starts the backup service on power on if it was missed while
the system was powered down, for example.

###### Motivation for this change
Adds an option to make a backup job start on boot if it was missed during power off. This is useful for backups on personal machines where you might have the machine off during the normally scheduled backup.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS (n/a)
   - [ ] other Linux distributions (n/a)
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) (tested the service to make sure it runs)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blargg** [comment]
This version follows along with how nix-gc allows for these parameters https://github.com/NixOS/nixpkgs/blob/83d907fd760d9ee4f49b4b7e4b1c6682f137b573/nixos/modules/services/misc/nix-gc.nix#L50.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow our commit naming guide: ``nixos/borgbackup: Add a persistent option``.

File: nixos/modules/services/backup/borgbackup.nix [hunk 1]
```diff
@@ -669,6 +689,10 @@ in {
         # A repo named "foo" is mapped to systemd.services.borgbackup-repo-foo
         // mapAttrs' mkRepoService repos;
 
+      systemd.timers =
+        # A job named "foo" is mapped to systemd.timers.borgbackup-job-foo
+        mapAttrs' mkBackupTimers jobs;
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/borgbackup.nix:694]:
```suggestion
      # A job named "foo" is mapped to systemd.timers.borgbackup-job-foo
      systemd.timers = mapAttrs' mkBackupTimers jobs;
```

Comment (blargg) [reply nixos/modules/services/backup/borgbackup.nix]: Thanks, applied this change.

**Comment by blargg** [comment]
Updated the formatting and the commit name.

**Comment by blargg** [comment]
Not sure if we want to act on the following as part of this PR, but will mention this here either way:
Since we add the ability to start on boot, it's a lot more likely that we run into the case where you try to back up to a remote server before the network starts up, causing the run to fail.

It seems like this is fixable by adding
```
wants = [ "network-online.target" ];
after = [ "network-online.target" ];
```
to either the timer or service. I think we could automatically detect remote usage by checking for "@" in the string for the repo, and possibly provide a setting to override.

**Comment by Stunkymonkey** [comment]
there is a previous PR #75056. I would prefer calling it `persistentTimer`, to point to the relationship with the timer unit (like @mweinelt suggested).

The rest looks fine to me.

File: nixos/modules/services/backup/borgbackup.nix [hunk 2]
```diff
@@ -98,6 +98,13 @@ let
       inherit (cfg) startAt;
     };
 
+    mkBackupTimers = name: cfg:
+      nameValuePair "borgbackup-job-${name}" {
```
Comment (Stunkymonkey) [inline nixos/modules/services/backup/borgbackup.nix:102]: adding a description would be nice as well

**Review by Stunkymonkey — CHANGES_REQUESTED** [review]

**Comment by Stunkymonkey** [comment]
@mweinelt is setting the default value to `true` desirable?

**Comment by Stunkymonkey** [comment]
fixed in #153346

**Pull request closed without merging**